### PR TITLE
Fixing the disappearing slider when defaults set

### LINKED ISSUE #725: Resources Accessed: Message displays saying no data is available when the user changes their setting
# Expected behavior (A clear and concise description of what you expected to happen) : 
User can select a new setting from the filter drop-down and the filtered results and week timeline continue to display correctly when the setting is saved.

## Describe the bug (Tell us what happens instead of the expected behavior) :
After clicking Save My Setting, 2 unexpected results are happening: 
1. The timeline disappears from the top of the page  
2. The filtered list of results disappears and a message displays "Resource data for your selections is not available.  (even though results are available)

Workaround: Refresh the page.

## Steps to Reproduce :
1.  User goes to Resources page
2. User selects a filter on the Resources page (make sure resources still show in the filtered list)
3. User clicks Save My Setting

### PR DESCRIPTION
**Pull request by pushyamig**
Fixes #725
1. I have refactored the code and used the same naming convention for hook as in AssignmentPlanning view for making it easier to understand login in both views.
2. Added the Snackbar, separated the get/set default logic outside of `ResourcesAccessed.js` as in `AssignmentPlanning.js`
3. once thing I did not do is the content call is still inline in `useEffect` instead of `api.js`. I felt it can be handled separately as the logic is different from Assignment view

### CODE REVIEW

File: assets/src/containers/ResourcesAccessed.js [hunk 1]
```diff
@@ -236,7 +229,6 @@ function ResourcesAccessed (props) {
       )
     }
   }
-  if (error) return (<WarningBanner/>)
```
Comment (pushyamig) [inline assets/src/containers/ResourcesAccessed.js]: Removed this as it is not needed, when we put the content call in `api.js` then we can handle this. The BE always returns empty response so logic is baked in to display appropriate message to user

File: assets/src/containers/AssignmentPlanning.js [hunk 2]
```diff
@@ -84,6 +84,7 @@ function AssignmentPlanning (props) {
     if (userSettingLoaded) {
       if (isObjectEmpty(userSetting.default)) {
         setAssignmentGradeFilter(0)
+        setUserSavedFilterSetting(0)
```
Comment (pushyamig) [inline assets/src/containers/AssignmentPlanning.js]: During the initial load of the assignment view, the display needs to say `My Current Setting` instead it was showing ` `Remember My Setting`, this fixes that

File: assets/src/containers/ResourcesAccessed.js [hunk 3]
```diff
@@ -99,41 +106,28 @@ function ResourcesAccessed (props) {
...
-            setDefaultValue(gradeRangeFilter)
-            return
-          }
-          setDefaultLabel(settingNotUpdated)
-        }).catch(err => {
-        setDefaultLabel(settingNotUpdated)
-      })
+  // if user setting is saved, don't show checkbox and sync userSavedFilterSetting with assignmentGradeFilter
+  useEffect(() => {
+    console.log()
```
Comment (justin0022) [inline assets/src/containers/ResourcesAccessed.js]: Other than the `console.log()` looks good to me

File: assets/src/containers/ResourcesAccessed.js [hunk 4]
```diff
@@ -99,41 +106,28 @@ function ResourcesAccessed (props) {
...
+    if (userSettingSaved) {
+      setShowSaveSetting(false)
+      setUserSavedFilterSetting(resourceGradeFilter)
     }
-  }
+  }, [userSettingSaved])
 
   useEffect(() => {
+    console.log("Setting the slider range")
```
Comment (justin0022) [inline assets/src/containers/ResourcesAccessed.js]: Other than the `console.log()` looks good to me

File: assets/src/containers/ResourcesAccessed.js (hunk 3 above)
Comment (pushyamig) [reply assets/src/containers/ResourcesAccessed.js]: Done

File: assets/src/containers/ResourcesAccessed.js (hunk 4 above)
Comment (pushyamig) [reply assets/src/containers/ResourcesAccessed.js]: Done

**Pull request merged**